app/src/main/java/com/vasilevkin/catsanddogs/features/animalList/presenter/MainPresenter.kt

```diff
@@ -0,0 +1,93 @@
+package com.vasilevkin.catsanddogs.features.animalList.presenter
+
+import android.app.Activity
+import com.vasilevkin.catsanddogs.delegateadapter.diff.IComparableItem
+import com.vasilevkin.catsanddogs.features.animalList.IMainContract
+import com.vasilevkin.catsanddogs.models.localModels.Animal
+import com.vasilevkin.catsanddogs.models.localModels.BigViewpagerLocalModel
+import com.vasilevkin.catsanddogs.models.localModels.LongHorizontalCatLocalModel
+import com.vasilevkin.catsanddogs.models.localModels.SquareCatLocalModel
+import com.vasilevkin.catsanddogs.repository.IAnimalRepository
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import io.reactivex.internal.schedulers.IoScheduler
+
+
+class MainPresenter(
+    view: IMainContract.View,
+    private val animalRepository: IAnimalRepository
+) : IMainContract.Presenter {
+
+    private var view: IMainContract.View? = view
+
+    private var disposable: Disposable? = null
+
+    // IMainContract methods
+
+    override fun onViewCreated() {
+        loadAnimals()
+    }
+
+    override fun onLoadMoreAnimalsTapped() {
+        loadAnimals()
+    }
+
+    override fun onDestroy() {
+        this.view = null
+        this.disposable?.dispose()
+    }
+
+    // Private methods
+
+    private fun loadAnimals() {
+        val animals = animalRepository.getAllAnimals()
+
+        this.disposable = animals
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribeOn(IoScheduler())
+            .subscribe { cats ->
+                val objects = ArrayList<IComparableItem>(20)
+                for (i in cats.indices) {
+                    var item: IComparableItem?
+
+                    when (i % 6) {
```

Comment:
hardcoded value 6

Reply:
Fixed, but check how I fixed it, may be there is a better way?